extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -1019,7 +1024,12 @@ class LifecycleTask extends BackbeatTask {
                 .setAttribute('target.bucket', bucketData.target.bucket)
                 .setAttribute('target.accountId', bucketData.target.accountId)
                 .setAttribute('target.key', obj.Key)
-                .setAttribute('details.lastModified', obj.LastModified);
+                .setAttribute('details.lastModified', obj.LastModified)
+                .setAttribute('details.dataStoreName', obj.StorageClass || '')
```

Comment:
in case this is 'STANDARD', the dataStoreName should actually be the bucket's `region`... we don't currently have access to it, but it may be retrieved through a simple `HEAD` on the bucket.